src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -10490,6 +10490,115 @@ public void SynthesizedDelegateTypes_NamedArguments_MoreThan16Parameters()
                 Diagnostic(ErrorCode.ERR_NoCorrespondingArgument, "inferred").WithArguments("arg1", "<anonymous delegate>").WithLocation(7, 1));
         }
 
+        [Fact]
+        public void SynthesizedDelegateTypes_MoreThan16Parameters_DefaultParameterValue()
+        {
+            var range = Enumerable.Range(1, 17);
```

Comment:
>Enumerable.Range(1, 17); [](http://example.com/codeflow?start=24&length=24)

Is scenario with less number of parameters not interesting?

Reply:
Such scenarios are tested elsewhere. With more than 16 parameters, `Func` cannot be used anymore, so synthesized delegate are used (even before this PR). This test shows how that combines with default parameters where synthesized delegates are also used after this PR.